parser/src/powdr.lalrpop

```diff
@@ -514,7 +514,7 @@ BinaryAndOp: BinaryOperator = {
 }
 
 BitShift: Box<Expression> = {
-    BitShift BitShiftOp Sum => Box::new(Expression::BinaryOperation(<>)),
+    <start:@L> <left:BitShift> <op:BitShiftOp> <right:Sum> => Box::new(Expression::BinaryOperation(ctx.source_ref(start), BinaryOperation { left, op, right })),
```

Comment:
I'm wondering if we could simplify this into
```suggestion
    <start:@L> <left:BitShift> <op:BitShiftOp> <right:Sum> => Box::new((BinaryOperation { left, op, right }).into().with_source(ctx.source_ref(start))),
```

if we have Into for conversion between the Expression sub-structs into Expression (and unknown source) and then a generic function on Expression:

```
impl Expression {
  pub fn with_source(&mut self, s: SourceRef) -> Self {
    *(self.source_mut()) = s;
    self
  }
}
```